llvm/lib/Target/RISCV/RISCVISelLowering.cpp

```diff
@@ -13670,6 +13671,14 @@ static bool narrowIndex(SDValue &N, ISD::MemIndexType IndexType, SelectionDAG &D
   return true;
 }
 
+// We're performing 2 combinations here:
+// === Rule 1 ===
+// Given VFirst = (vfirst_vl ..., EVL) and
+// (seteq (riscv_selectLT VFirst, 0, EVL, VFirst), EVL), we can replace it with
+// (setLT VFirst, 0). Similar replacements are done for variants w/ `setne` and
```

Comment:
Is this pattern only generated by `RISCVTargetLowering::lowerVPCttzElements`? Would it be easier to just emit (setLT VFirst, 0) there

Reply:
The outer seteq/setne is not generated from `RISCVTargetLowering::lowerVPCttzElements`. Only `(riscv_selectLT VFirst, 0, EVL, VFirst)` comes from `RISCVTargetLowering::lowerVPCttzElements`